### PR TITLE
modulesync 2.5.1 and drop Puppet 4

### PR DESCRIPTION
**Pull request by bastelfreak**
modulesync 2.5.1

### CODE REVIEW

**Comment by ekohl** [comment]
Now has a merge conflict

File: spec/defines/selinux_exec_restorecon_spec.rb [hunk 1]
```diff
@@ -2,10 +2,14 @@
 
 describe 'selinux::exec_restorecon' do
   on_supported_os.each do |os, _facts|
+
+
     context "on #{os}" do
       context 'with resource titled /opt/mycustompath' do
         let(:title) { '/opt/mycustompath' }
-
+        let :facts do
```
Comment (ekohl) [inline spec/defines/selinux_exec_restorecon_spec.rb]: You can do this at a higher level (just below `context "on #{os}" do`) and just do it once

File: spec/default_module_facts.yml [hunk 2]
```diff
@@ -1,7 +1,4 @@
 ---
-osfamily: RedHat
-operatingsystem: RedHat
-operatingsystemmajrelease: '7'
 # concat facts
 id: 0
```
Comment (ekohl) [inline spec/default_module_facts.yml]: This and path can also be dropped, as well as the comment

File: manifests/params.pp [hunk 3]
```diff
@@ -5,7 +5,7 @@
 class selinux::params {
   $module_build_root = "${facts['puppet_vardir']}/puppet-selinux"
 
-  case $facts['osfamily'] {
+  case $facts['os']['family'] {
     'RedHat': {
       if $facts['operatingsystem'] == 'Amazon' {
```
Comment (ekohl) [inline manifests/params.pp]: If you want to use the modern facts, then this should probably also be replaced (as well as the one below).

File: spec/classes/selinux_package_spec.rb [hunk 4]
```diff
@@ -9,7 +9,8 @@
             osfamily: 'RedHat',
```
Comment (ekohl) [inline spec/classes/selinux_package_spec.rb]: You can drop the old facts here if you want.

File: manifests/params.pp [hunk 5]
```diff
@@ -5,12 +5,12 @@
...
-  case $facts['osfamily'] {
+  case $facts['os']['family'] {
     'RedHat': {
-      if $facts['operatingsystem'] == 'Amazon' {
+      if $facts['os']['name'] == 'Amazon' {
         $package_name = 'policycoreutils'
       } else {
-        $package_name = $facts['operatingsystemmajrelease'] ? {
+        $package_name = $facts['os']['release']['major'] ? {
           '5'     => 'policycoreutils',
```
Comment (ekohl) [inline manifests/params.pp]: Do you need to convert this to integers or are they still strings? I think the type should match in the specs.

Comment (bastelfreak) [reply manifests/params.pp]: that isn't required. Also facter displays it as string, not integer.

**Comment by bastelfreak** [comment]
@ekohl IMO merging this and doing a major release is fine, that's wha we agreed on in https://github.com/voxpupuli/voxpupuli.github.io/pull/157

**Comment by ekohl** [comment]
@bastelfreak I'd prefer to do a release, bump major and then wait for other changes to come in

**Comment by dhollinger** [comment]
Based on our conversation in https://github.com/voxpupuli/voxpupuli.github.io/pull/157, we wanted to not ever do a release just for removing puppet 4. As such, I'm ok with this PR, @ekohl. Doing a minor/patch release to prepare for a major release change that removes Puppet 4 is exactly what the community agreed to avoid doing.

**Pull request merged**